pkg/backends/locust/backend.go

```diff
@@ -206,8 +206,7 @@ func (b *Backend) SyncStatus(ctx context.Context, loadTest loadTestV1.LoadTest,
 		loadTestStatus.Phase = loadTestV1.LoadTestCreating
 	}
 
-	if loadTestStatus.Phase == loadTestV1.LoadTestErrored ||
-		loadTestStatus.Phase == loadTestV1.LoadTestFinished {
```

Comment:
I don't think this should be removed; once loadtest is marked as Finished, it's over and its status should not be updated again.

I believe the addition in L242 already fixes the bug where the loadtest is not being deleted.

Reply:
Since `Job Status` is not persisting on the LoadTest object, it is not saved after we finish processing this LoadTest object in syncHandler. L242 updates the status but it turns to `nill` as soon as we finish processing this object. And the next time controller picks this object, it will drop it from SyncStatus as finished, and `checkLoadTestLifeTimeExceeded` will drop it too, because `loadTest.Status.JobStatus.CompletionTime == nil`. And that's how this object gets stuck in the cluster forever.  
<img width="394" alt="Screenshot 2021-06-02 at 17 36 23" src="https://user-images.githubusercontent.com/21171767/120509827-24e8a000-c3c9-11eb-83ed-941d72c25dcd.png">

Reply:
Looks like we need to find out why `LoadTest.Status.JobStatus` is nil on every `syncHandler` iteration, but that is a separate issue.